### PR TITLE
Add unit tests for db/repos/migrations - Closes #1589

### LINKED ISSUE #1589: Add unit test coverage for db/repos/migrations
Parent: #1484

### Expected behavior
We need 100% coverage for db/repos/migrations

### Actual behavior
Currently there is no coverage for this repo.

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?
No unit test coverage for `db/repos/migrations`

### How did I fix it?
Added test cases thoroughly for every method in db/repos/migrations. And got 100% coverage for that file.

### How to test it?

```
npm mocha test/unit/db/repos/migrations.js
```

### Review checklist

* The PR solves #1589
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Comment by vitaly-t** [comment]
I hope all that usage of `yield` is working as you think it is...

The thing is, their use inside tests like this is not the same as their use within `pg-promise` methods `task` and `tx`. Those methods provide an automatic adapter that converts `yield` for promises into native `yield` results. See https://github.com/vitaly-t/pg-promise/blob/master/lib/async.js

And so, hence my question - does it really work? The only way it works right, is if the test framework provides the same type of adapter.

**Comment by nazarhussain** [comment]
@vitaly-t Yes these `yield` works perfectly fine in tests and is enabled by a mocha plugin `co-mocha`

File: test/unit/db/repos/migrations.js [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+
+			it('should resolve with empty array if there is no pending migration', function*() {
+				const pending = yield db.migrations.readPending(
+					fileIds[fileIds.length - 1]
+				);
+
+				return expect(pending).to.be.empty;
+			});
+
+			it('should resolve the list in correct format', function*() {
```
Comment (4miners) [inline test/unit/db/repos/migrations.js]: `resolve with list`

File: test/unit/db/repos/migrations.js [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+						fileNames = files.map(f => f.match(/(\d+)_(.+).sql/)[2]);
+
+						return db.migrations.readPending(fileIds[0]);
+					})
+					.then(_updates => {
+						updates = _updates;
+					});
+			});
+
+			it('should apply all pending migrations in a transaction', () => {
```
Comment (4miners) [inline test/unit/db/repos/migrations.js]: `in a transaction`, mean all migrations in single transaction or all migrations in separate transactions (one per migration)? specify please

File: test/unit/db/repos/migrations.js [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+						applyRuntime: sinonSandbox.stub().resolves(true),
+						readPending: sinonSandbox.stub().resolves(updates),
+					},
+					tx: sinonSandbox.stub().callsArgWith(1, t2),
+				};
+
+				// Since sinon can only call a callback or resolve to promise
+				// but pg-promise db.tx is mix, it accepts a callback and returns a promise
+				// and there is no way in sinon to resolve to to value return of a callacbk
+				// so have to hack it by delaying the execution of resolve untill callback perform
```
Comment (4miners) [inline test/unit/db/repos/migrations.js]: `until callback is called`

File: test/unit/db/repos/migrations.js (hunk 2 above)
Comment (nazarhussain) [reply test/unit/db/repos/migrations.js]: In one transaction as main and nested save points for each individual migration.

**Comment by karmacoma** [comment]
Reopened: https://github.com/LiskHQ/lisk/issues/1589#issuecomment-368637479

**Pull request merged**